datahub-web-react/src/app/lineage/utils/columnLineageUtils.ts

```diff
@@ -82,9 +81,33 @@ export function populateColumnsByUrn(
                     convertInputFieldsToSchemaFields(fetchedEntity.inputFields) as SchemaField[],
                 ),
             };
+        } else if (fetchedEntity.type === EntityType.DataJob && fetchedEntity.fineGrainedLineages) {
+            // Add upstream fields from fineGrainedLineage onto DataJob to mimic upstream dataset fields.
+            // DataJobs will virtually "have" these fields so we can draw full column paths
+            // from upstream dataset fields to downstream dataset fields.
+            const fields: SchemaField[] = [];
+            fetchedEntity.fineGrainedLineages.forEach((fineGrainedLineage) => {
+                fineGrainedLineage.upstreams?.forEach((upstream) => {
+                    fields.push({
```

Comment:
Do we de-duplicate in the case where same field serves as the input twice?

Reply:
oh no good point. I don't think having duplicates in here would break anything since the real goal of this columns by urn is to ensure we know which columns are on which entities, but having duplicates isn't intended and could happen if an upstream column is in two different `fineGrainedLineage` objects. will do a check before pushing onto `fields`.

Reply:
sounds good!